src/dmd/cparse.d

```diff
@@ -1500,15 +1525,25 @@ final class CParser(AST) : Parser!AST
                 {
                     if (hasInitializer)
                         error("no initializer for function declaration");
-                    AST.Declaration s = new AST.FuncDeclaration(token.loc, Loc.initial, id, SCWtoSTC(level, scw), dt);
-                    symbols.push(s);
+                    s = new AST.FuncDeclaration(token.loc, Loc.initial, id, SCWtoSTC(level, scw), dt);
                 }
                 else
                 {
-                    AST.Declaration s = new AST.VarDeclaration(token.loc, dt, id, initializer, SCWtoSTC(level, scw));
-                    symbols.push(s);
+                    s = new AST.VarDeclaration(token.loc, dt, id, initializer, SCWtoSTC(level, scw));
                 }
             }
+            if (s !is null)
+            {
+                if (asmname)
+                {
```

Comment:
Explain what is being created in this section.

Reply:
#12555